src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -128,12 +129,14 @@ func adminAddressesFromK8S(ctx context.Context, namespace string) ([]string, err
 		return nil, err
 	}
 
+	clusterDomain := getClusterDomain()
 	// Get the admin addresses from ContainerPort.
 	var adminAddresses []string
 	for _, p := range pods.Items {
 		for _, port := range p.Spec.Containers[0].Ports {
 			if port.Name == "admin" {
-				a := fmt.Sprintf("%v:%v", p.Status.PodIP, port.ContainerPort)
+				fqdn := fmt.Sprintf("%v.%v.%v.svc.%v", p.Spec.Hostname, p.Spec.Subdomain, p.Namespace, clusterDomain)
```

Comment:
pods don't have a spec.hostname or spec.subdomain, so this has me confused.

Reply:
I used this because of the go API docs:

```go
// Specifies the hostname of the Pod
// If not specified, the pod's hostname will be set to a system-defined value.
// +optional
Hostname string `json:"hostname,omitempty" protobuf:"bytes,16,opt,name=hostname"`
// If specified, the fully qualified Pod hostname will be "<hostname>.<subdomain>.<pod namespace>.svc.<cluster domain>".
// If not specified, the pod will not have a domain name at all.
// +optional
Subdomain string `json:"subdomain,omitempty" protobuf:"bytes,17,opt,name=subdomain"`
```
That's why I used `subdomain` instead of looking for the service name in another query.

And https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields

> Currently when a Pod is created, its hostname (as observed from within the Pod) is the Pod's metadata.name value

Also ran the following in a helm deployment:
```
$ k get pod -n redpanda redpanda-0 -o json | jq '.spec.hostname'
"redpanda-0"
$ k get pod -n redpanda redpanda-0 -o json | jq '.spec.subdomain'
"redpanda"
```

However, this is an optional value so I'm not entirely sure if this will always be the case

Reply:
Changed this to use service name, and list the pods based on the service label selector.